tensorboard/webapp/app_routing/effects/app_routing_effects.ts

```diff
@@ -384,9 +387,17 @@ export class AppRoutingEffects {
     );
 
     return changeUrl$.pipe(
-      withLatestFrom(this.store.select(getActiveRoute)),
-      map(([{route}, oldRoute]) => {
-        return navigated({before: oldRoute, after: route});
+      withLatestFrom(
+        this.store.select(getActiveRoute),
+        this.store.select(getActiveNamespaceId)
+      ),
+      map(([{route}, oldRoute, oldNamespaceId]) => {
+        return navigated({
+          before: oldRoute,
+          after: route,
+          beforeNamespaceId: oldNamespaceId,
```

Comment:
app_routing_effects gets beforeNamespaceId from state and generates the afterNamespaceId from routeId.